### DISCUSSION TITLE
Petition to change that gnarly looking loading-spinner.gif

### DISCUSSION THREAD

**Discussion by goellner** [craftcms/cms#9109] [Ideas] [answered]
# Petition to change loading-spinner.gif

## The problem
Every time we log in to Craft CMS or it has to do it's thing in the background, our eyes have to experience one of the internets most copied loading spinners with its artefacts. 

![my-eyes-hurt](https://user-images.githubusercontent.com/3986748/92114594-62e07580-edf1-11ea-9d07-dc69007a6d5f.png)

## What do we, the people, want?
We, the people, want Pixel & Tonic to give our eyes a nicer looking loading-spinner.gif. Let us enjoy crisp edges, while we wait for stuff to happen. 

## When do we want it?
We can work with the **next patch release** or any release **before** that.

**Comment by engram-design** [comment]
Here's one ready to go: 
![Sep-04-2020 08-54-50](https://user-images.githubusercontent.com/1221575/92181369-64ab4700-ee8c-11ea-9c3d-bb16dcc8e36a.gif)

```
@keyframes loading {
    0% {
        transform: rotate(0)
    } 100% {
        transform: rotate(360deg)
    }
}

.spinner {
    position: relative;
    pointer-events: none;
    color: transparent !important;
    min-height: 1rem;
}

.spinner:after {
    position: absolute;
    display: block;
    height: 1rem;
    width: 1rem;
    margin-top: -.5rem;
    margin-left: -.5rem;
    border-radius: 9999px;
    animation: loading .5s linear infinite;
    border-color: transparent transparent #e5422b #e5422b;
    border-style: solid;
    border-width: 2px;
    content: "";
    left: 50%;
    top: 50%;
    z-index: 1;
}

.btn.submit.spinner {
    color: transparent !important;
}
```

Pros:
- No additional HTTP requests to fetch gif files
- Responsive and retina-ready
- Flexible - change the size and color in CSS
- Can be used on buttons `.btn .spinner` and replaces text with a spinner

**Comment by brandonkelly** [comment]
Been meaning to pull the plugin store spinner into the rest of the CP.

Pretty sure that horrendous thing in the OP is not our own CSS though!

**Comment by goellner** [comment]
If the plugin store spinner is the same one as the update spinner, we should talk about that one too: 
![spinner](https://user-images.githubusercontent.com/3986748/92360700-8b1bed00-f0ed-11ea-84ad-81ce39a100de.gif)

**Comment by brandonkelly** [comment]
Nope, was referring to this one:

<img alt="The loading spinner used by the Plugin Store" src="https://user-images.githubusercontent.com/47792/92491362-fe356a00-f1a6-11ea-84b5-feaa0deda8ad.png" width="251">

**Comment by nickdunn** [comment]
> If the plugin store spinner is the same one as the update spinner, we should talk about that one too:
> ![spinner](https://user-images.githubusercontent.com/3986748/92360700-8b1bed00-f0ed-11ea-84ad-81ce39a100de.gif)

Oh the antialiased edged on the wrong background colour make me so sad here. Is this gif used in different places on different backgrounds?

**Comment by brandonkelly** [comment]
Just fixed the look of spinners when positioned over the BG color:

![A spinner graphic](https://user-images.githubusercontent.com/47792/106048942-5dcedc00-609a-11eb-9ec0-f0159a23ec09.png)

**Comment by brandonkelly** [comment, marked as answer]
Swapped to the Plugin Store-style spinners everywhere in the control panel, for Craft 4 (aa9ff3d9eaaaef811c6e69c0957d8866a1f65de3) ✨


<img src="https://user-images.githubusercontent.com/47792/152693046-718891f6-f52c-41c6-ad7c-8307c48421c2.gif" width="130" height="75" alt="Screen recording of a Save button, whose “Save” label is replaced with a loading spinner while the save operation is completed.">
